From : http://launchpadlibrarian.net/439818280/cfitsio_3.450-3_3.470-3.diff.gz
Auther : Aurelien Jarno <aurel32@debian.org>
2019-08-01

--- a/fitsio2.h	2018-05-10 03:16:01.000000000 +0800
+++ b/fitsio2.h	2021-12-30 15:19:57.119440200 +0800
@@ -137,6 +137,20 @@
 #    error "can't handle long size given by _MIPS_SZLONG"
 #  endif
 
+#elif defined(__riscv)
+
+/* RISC-V is little endian */
+
+#define BYTESWAPPED TRUE
+
+#  if __riscv_xlen == 32
+#    define LONGSIZE 32
+#  elif __riscv_xlen == 64
+#    define LONGSIZE 64
+#  else
+#    error "can't handle long size given by __riscv_xlen"
+#  endif
+
 /* ============================================================== */
 /*  the following are all 32-bit byteswapped platforms            */
 
